File: Hello/step07sqlite/src/main/java/com/example/step07sqlite/DBHelper.java
package com.example.step07sqlite;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

/*
    DB 생성 및 리셋을 도와주는 도우미 클래스 만들기
    - SQLiteOpenHelper 추상 클래스를 상속 받아서 만든다.
 */
public class DBHelper extends SQLiteOpenHelper {
    // 생성자
    public DBHelper(@Nullable Context context, @Nullable String name, @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }

    // App 에서 DB 를 처음 사용할 때 호출되는 메소드
    @Override
    public void onCreate(SQLiteDatabase db) {
        // 사용할 테이블을 만들면 된다.
        String sql = "CREATE TABLE todo " +
                "(NUM INTEGER PRIMARY KEY AUTOINCREMENT, " +
                "CONTENT TEXT, REGDATE TEXT)";
        // SQLiteDataBase 객체를 이용해서 실행한다.
        db.execSQL(sql);
    }

    // DBHelper 객체를 생성할 때 vesion 숫자가 올라가면(변경되면) 호출되는 메소드
    @Override
    public void onUpgrade(SQLiteDatabase db, int i, int i1) {
        // 업그레이드 할 내용을 작성하면 된다.
        db.execSQL("DROP TABLE IF EXISTS todo");
        // 다시 만들어 질 수 있도록 onCreate() 메소드를 호출한다.
        onCreate(db);
    }
}

File: Hello/step07sqlite2/src/main/java/com/example/step07sqlite2/DBHelper.kt
package com.example.step07sqlite2

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

/*
    SQLiteOpenHelper 클래스는 추상 클래스이기 때문에 추상 메소드를 오버라이드 해야한다.
    DBHelper 클래스의 대표 생성자에서 부모 생성자에 전달할 값을 받아야 한다.
    SQLiteOpenHelper 클래스의 생성자에 필요한 값을 전달해야 한다.
 */
class DBHelper constructor(
    context : Context?,
    name: String?,
    factory: SQLiteDatabase.CursorFactory,
    version: Int
) : SQLiteOpenHelper(context, name, factory, version) {

    init {

    }

    override fun onCreate(db: SQLiteDatabase?) {
        val sql = """
            CREATE TABLE todo 
            (NUM INTEGER PRIMARY KEY AUTOINCREMENT, CONTENT TEXT, REGDATE TEXT)
        """.trimIndent()
        db!!.execSQL(sql)
    }

    override fun onUpgrade(db: SQLiteDatabase?, i: Int, i1: Int) {
        db?.execSQL("DROP TABLE IF EXISTS todo")
        onCreate(db)
        db.execSQL()
    }
}